src/actions/MatrixActionCreators.ts

```diff
@@ -58,12 +58,17 @@ function createSyncAction(matrixClient: MatrixClient, state: string, prevState:
  * @param {MatrixEvent} accountDataEvent the account data event.
  * @returns {AccountDataAction} an action of type MatrixActions.accountData.
  */
-function createAccountDataAction(matrixClient: MatrixClient, accountDataEvent: MatrixEvent): ActionPayload {
+function createAccountDataAction(
+    matrixClient: MatrixClient,
+    accountDataEvent: MatrixEvent,
+    previousAccountDataEvent?: MatrixEvent,
+): ActionPayload {
     return {
         action: "MatrixActions.accountData",
         event: accountDataEvent,
         event_type: accountDataEvent.getType(),
         event_content: accountDataEvent.getContent(),
+        previousEvent: previousAccountDataEvent,
```

Comment:
add this to the definition of `AccountDataAction` please.

Reply:
It feels a bit asymmetric that we add the type and content of `accountDataEvent` to the action separately, but use a single field for `previousAccountDataEvent`. Did you consider splitting `previousAccountDataEvent` for symmetry?

Reply:
I don't really like the destructuring in the first place as the `event` is also included in full.
Most other matrix actions that have events in their payload don't do any destructuring.
`previousEvent` will also always have the same type as event